### PR TITLE
Deploy to development on pushes to main

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

Closes https://github.com/cal-itp/reports/issues/176
Fixes https://github.com/cal-itp/reports/issues/115

I've updated the top-level repo README to include a description of the new deploys. I've left auto-deploys enabled in Netlify, but they won't trigger unless we push commits to the `production` branch.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation

## How has this been tested?
See development branch Actions.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -14,40 +15,31 @@ jobs:
...
     - uses: google-github-actions/setup-gcloud@v0
       with:
         service_account_key: ${{ secrets.GCP_SA_KEY }}
         export_default_credentials: true
 
-    # development
+    # copy data ----
     - name: Copy development report data
-      if: ${{ github.ref == 'refs/heads/development' }}
+      if: ${{ github.event_name == 'push' }}
```
Comment (acouch) [inline .github/workflows/main.yml:38]: Just to clarify. Any push to main will update the dev site, and a new tag will push to prod?

**Review by lauriemerrell — COMMENTED** [review]
Can you add a little more clarification to the PR body about why this is being done incrementally / what the next steps are? One of the specific ACs in https://github.com/cal-itp/reports/issues/176 is documentation of how this works, I understand if that is planned for later but I am nervous about making changes here without clarifying when that documentation will be ready/when the full workflow will be actually ready.

Related: Have the settings been changed on the Netlify side? 

Basically just wondering if this PR will leave it in a state where it's impossible to deploy to prod?

File: .github/workflows/main.yml (hunk 1 above)
Comment (atvaccaro) [reply .github/workflows/main.yml:38]: Yes that's the ultimate goal.

**Comment by atvaccaro** [comment]
> Can you add a little more clarification to the PR body about why this is being done incrementally / what the next steps are? One of the specific ACs in #176 is documentation of how this works, I understand if that is planned for later but I am nervous about making changes here without clarifying when that documentation will be ready/when the full workflow will be actually ready.
> 
> Related: Have the settings been changed on the Netlify side?
> 
> Basically just wondering if this PR will leave it in a state where it's impossible to deploy to prod?

I've added to the README, and I did leave Netlify continuous deploys enabled if we want to revert the CI/CD changes.

File: README.md [hunk 2]
```diff
@@ -154,35 +154,30 @@ and open up a web browser, and navigate to:
 ### Pushing to google cloud - Development
 
 The next step is to update the development bucket in google cloud with the new data.
-
-```shell
-gsutil -m rsync -r outputs/ gs://gtfs-data-test/report_gtfs_schedule/
-```
-
 In the case where data must be overwritten (please use caution!) a `-d` flag can be added to the above command:
```
Comment (lauriemerrell) [inline README.md:157]: if above command removed, can we refactor this sentence to not reference it?

**Review by lauriemerrell — COMMENTED** [review]
one inline nit & I don't think that using `Closes #176 and #115` appropriately links both issues, can you attach #115 as fixed? otherwise LGTM but I'm not a repo owner

File: README.md (hunk 2 above)
Comment (edasmalchi) [reply README.md:157]: +1

**Review by edasmalchi — CHANGES_REQUESTED** [review]
Looks like the readme needs a small tweak as @lauriemerrell pointed out, otherwise lgtm!

File: README.md (hunk 2 above)
Comment (edasmalchi) [reply README.md:157]: Actually just looked at the rich diff and the command is still in, just _below_ instead of above! So swapping those 2 words oughta be enough

Comment (atvaccaro) [reply README.md]: I altered the other example with an optional `-d` flag so I think the new language should cover both places.

**Review by edasmalchi — APPROVED** [review]
lgtm, thanks!
